src/barnetilsyn/steg/6-barnepass/hjelper.ts

```diff
@@ -0,0 +1,92 @@
+import { subYears } from 'date-fns';
+import { IBarn } from '../../../models/barn';
+import {
+  dagensDato,
+  erPeriodeGyldig,
+  strengTilDato,
+} from '../../../utils/dato';
+import { IBarnepassOrdning } from '../../models/barnepass';
+import { ISøknad } from '../../models/søknad';
+import { ESøkerFraBestemtMåned } from '../../../models/steg/dinsituasjon/meromsituasjon';
+import { harValgtSvar } from '../../../utils/spørsmålogsvar';
+
+export const harBarnAvsluttetFjerdeKlasse = (barn: IBarn): boolean => {
+  const { alder, født, fødselsdato } = barn;
+  const juniEllerFør = dagensDato.getUTCMonth() <= 6;
+
+  if (født?.verdi && parseInt(alder.verdi) >= 10)
+    if (juniEllerFør) {
+      return subYears(dagensDato, 11) >= strengTilDato(fødselsdato.verdi);
+    } else {
+      return subYears(dagensDato, 10) >= strengTilDato(fødselsdato.verdi);
```

Comment:
Vi ønsker ikke å sjekke datoer spesifikt her, kun om de er født ila et spesifikt år. Skal se om jeg klarer å komme opp med et bra forslag

Reply:
```
const harBarnAvsluttetFjerdeKlasse = (fødselsdato: string): boolean => {
    const fødselsår = strengTilDato(fødselsdato).getFullYear()
    const gjeldendeÅr = dagensDato.getFullYear()
    const juniEllerFør = dagensDato.getUTCMonth() < 6;
	if (juniEllerFør) {
		return fødselsår + 11 <= gjeldendeÅr
	} else {
		return fødselsår + 10 <= gjeldendeÅr
	}
};
```

Denne skal iallefall gi riktig resultater.

Eksempelbruk her;
https://www.typescriptlang.org/play/index.html?ssl=1&ssc=1&pln=2&pc=1#code/MYewdgzgLgBAJgQwOYFNIBEFRDAvDMFAdxkyhQAoByAJgAZ6BaOgNmYEYqBKAbgFgAUKEixoAJzRIAKgEsANmRz4KibAGUoEsEgBcMcTO1c9ZFHgB8MAN6CYMCVACuYsAWKkslVSA1akvQQBffgFBYWgYAAsEMQAhGLAAQQA3CDlHKHIoADEAKxQxOBQAaTkECAgzZQAzAA+4SrkIbz0DIz0AIxAQORQEV1xLGwE7O3DYOoaUJoBTsTx9TUlZBSwQCknG5rWuADpUHMc5OQBNPrEKLltR8ZgkfLkisCKAUfn8RFQMNf2UQ+OzjFLtcxuAIrlHGAZABRY4FbK1d7wZBoCCKX5QACqUgAwgBZcBQSKXGAAHhgLBCAEgZNUYBQIVDYb0xAixFxrIIqVSHM5XJtphA5jAANQwdjsMn4e7TJ6vMRcwIwQVmYbc3kuGAC2bzMXsOhSu4POUoN6KoIhQRhMGwOQyaBVGAAbVodDoAE5GOwaBwqAAaGCu-UcX0BoN0L0+uicMP0CUh6P+wNx9iR30AXR4lqEYJ6KF2chASAododuwAtggAA4UAAeFiiMXiLhSaQyWTyBSKpXKlTrXAHPCAA

Reply:
Hvis barnet er lagt til, så kan man velge å bruke fnr eller fødselsdato. 
Så da vil ikke fødselsdato i alle tilfeller eksistere 😣 

Dette er tilfeller som sjeldent vil inntreffe, men et case er når barnet akkurat er født/ikke er født ennå slik at man er nødt til å legge til barnet selv. Da blir søker nødt til å fylle inn enten fnr eller fødselsdato. 
La også merke til at alder ikke oppdateres ut ifra fnr i disse tilfellene. Så da får barnet `alder = 0` 🤔

Reply:
Du har rett i det.

Jeg har laget en fiks på at alder settes nå, men alder blir ikke rett å bruke her. Vi kan evt. bruke `navfaker` til å plukke ut riktig fødselsår ut fra _enten_ ident eller fødselsdato. Den er i en branch jeg sitter på nå og tar høyde for dnr.

Reply:
Enig om at alder blir ukorrekt å bruke. 

Det kan vi gjøre!